From 6b5b14200f32dd2139fda91ae3b26c258090a621 Mon Sep 17 00:00:00 2001
From: Alexey Veprev <alexey.veprev@intel.com>
Date: Thu, 4 Feb 2021 16:39:02 +0300
Subject: [PATCH 2/2] CMake: fix tbb4py installation

---
 python/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index da4f4f93..6c95bcde 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -49,8 +49,8 @@ add_test(NAME python_test
                  -DPYTHON_MODULE_BUILD_PATH=${PYTHON_BUILD_WORK_DIR}/build
                  -P ${PROJECT_SOURCE_DIR}/cmake/python/test_launcher.cmake)
 
-install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${PYTHON_BUILD_WORK_DIR}/build/lib/
-        DESTINATION ${CMAKE_INSTALL_LIBDIR}
+install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${PYTHON_BUILD_WORK_DIR}/build/
+        DESTINATION .
         COMPONENT tbb4py)
 
 if (UNIX AND NOT APPLE)
-- 
2.19.0.windows.1

