diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index ac38f996..98338bf2 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -484,7 +484,7 @@ if (MSVC AND CMAKE_VERSION VERSION_GREATER 3.13) # LINK_OPTIONS property first a
 endif()

 # TBB allocator tests
-if (NOT "${CMAKE_SYSTEM_PROCESSOR}" MATCHES "(armv7-a|aarch64|mips|arm64)")
+if (NOT "${CMAKE_SYSTEM_PROCESSOR}" MATCHES "mips")
     # Define TBB malloc tests
     tbb_add_test(SUBDIR tbbmalloc NAME test_scalable_allocator DEPENDENCIES TBB::tbbmalloc)
     tbb_add_test(SUBDIR tbbmalloc NAME test_malloc_pools DEPENDENCIES TBB::tbbmalloc)
