diff --git a/cmake/compilers/AppleClang.cmake b/cmake/compilers/AppleClang.cmake
index f5af4220..12f7b07b 100644
--- a/cmake/compilers/AppleClang.cmake
+++ b/cmake/compilers/AppleClang.cmake
@@ -20,7 +20,11 @@ set(TBB_WARNING_SUPPRESS -Wno-parentheses -Wno-non-virtual-dtor -Wno-dangling-el
 # For correct ucontext.h structures layout
 set(TBB_COMMON_COMPILE_FLAGS -D_XOPEN_SOURCE)

-set(TBB_MMD_FLAG -MMD)
+# Depfile options (e.g. -MD) are inserted automatically in some cases.
+# Don't add -MMD to avoid conflicts in such cases.
+if (NOT CMAKE_GENERATOR MATCHES "Ninja" AND NOT CMAKE_CXX_DEPENDS_USE_COMPILER)
+    set(TBB_MMD_FLAG -MMD)
+endif()

 # Ignore -Werror set through add_compile_options() or added to CMAKE_CXX_FLAGS if TBB_STRICT is disabled.
 if (NOT TBB_STRICT AND COMMAND tbb_remove_compile_flag)
