diff --git a/build/macos.inc b/build/macos.inc
index dde93216..2bdcde7e 100644
--- a/build/macos.inc
+++ b/build/macos.inc
@@ -36,11 +36,15 @@ ifndef arch
      export arch:=ppc32
    endif
  else
+  ifeq ($(shell /usr/sbin/sysctl -n hw.machine),arm64)
+   export arch:=arm64
+  else
    ifeq ($(shell /usr/sbin/sysctl -n hw.optional.x86_64 2>/dev/null),1)
      export arch:=intel64
    else
      export arch:=ia32
    endif
+  endif
  endif
 endif
 
