diff --git a/test/tbbmalloc/test_malloc_compliance.cpp b/test/tbbmalloc/test_malloc_compliance.cpp
index b65a22e7..cabba337 100644
--- a/test/tbbmalloc/test_malloc_compliance.cpp
+++ b/test/tbbmalloc/test_malloc_compliance.cpp
@@ -92,6 +92,8 @@ void limitMem( size_t limit )
     if (rlim.rlim_max==(rlim_t)RLIM_INFINITY)
         rlim.rlim_cur = (limit > 0) ? limit*MByte : rlim.rlim_max;
     else rlim.rlim_cur = (limit > 0 && static_cast<rlim_t>(limit)<rlim.rlim_max) ? limit*MByte : rlim.rlim_max;
+    REPORT("limit=%lu rlim_max=%lu rlim_cur=%lu RLIM_INFINITY=%lu\n", limit, rlim.rlim_max, rlim.rlim_cur, (rlim_t)RLIM_INFINITY);
+    REPORT("casted limit = %lu\n", static_cast<rlim_t>(limit));
     ret = setrlimit(RLIMIT_AS,&rlim);
     if (0 != ret) {
         REPORT("Can't set limits: errno %d\n", errno);
@@ -284,7 +286,7 @@ void ReallocParam()
         if (NULL == bufs[i])
             break;
     }
-    REQUIRE_MESSAGE(i<ITERS, "Limits should be decreased for the test to work.");
+    REQUIRE_MESSAGE(0, " Limits should be decreased for the test to work. " << i);

     Trealloc(bufs[0], 0);
     /* There is a race for the free space between different threads at
